### PR TITLE
Add scopes for transfer items based on received state

### PR DESCRIPTION
**Pull request by allisonlarson**
* Added fully received scope for transfer items whose received
 quantity equal the expected quantity
* Added partially received scope for transfer items whose received
 quantity is less than the expected quantity
* Specs

### CODE REVIEW

**Comment by athal7** [comment]
:+1:

**Comment by jhawthorn** [comment]
Should a `received_quantity` of 0 be considered "partially received"? Otherwise looks good.

**Comment by allisonlarson** [comment]
good point... Fixed @jhawthorn

**Comment by jhawthorn** [comment]
looks good :+1:

File: core/app/models/spree/transfer_item.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@ class TransferItem < ActiveRecord::Base
 
     scope :received, -> { where('received_quantity > 0') }
     scope :fully_received, -> { where('expected_quantity = received_quantity') }
+    scope :partially_received, -> { received.where('expected_quantity > received_quantity') }
```
Comment (gmacdougall) [inline core/app/models/spree/transfer_item.rb]: The values here aren't scoped to the table, which can potentially lead to issues in the future if someone joins another table which contains fields named `received_quantity` or `expected_quantity` and uses this scope. It's not super likely, but it's a good habit to get in to.

Compare to the SQL generated by:

```Ruby
scope :partially_received, -> {
  received.where(arel_table[:expected_quantity].gt(arel_table[:received_quantity]))
}
```

The generated SQL is:

```
("spree_transfer_items"."expected_quantity" > "spree_transfer_items"."received_quantity")
```
compared to
```
(expected_quantity > received_quantity)
```

**Comment by gmacdougall** [comment]
My comment is super minor, fine merging this as is.

:+1:

**Comment by allisonlarson** [comment]
@gmacdougall thats super good to know, I'll keep in mind for next time :smile:

File: core/app/models/spree/transfer_item.rb (hunk 1 above)
Comment (athal7) [inline core/app/models/spree/transfer_item.rb]: i agree, this would be a good change to make imho, we've run into this before

**Pull request merged**